src/extension/features/toolkit-reports/pages/income-vs-expense/component.jsx

```diff
@@ -56,10 +66,26 @@ export class IncomeVsExpenseComponent extends React.Component {
     collapsedSources: new Set(),
   };
 
+  _saveState(state) {
+    return JSON.stringify(Array.from([...state.collapsedSources]));
+  }
+
+  _parseState(storage) {
+    if (storage === null) {
+      return new Set();
+    }
+
+    return new Set(JSON.parse(storage));
+  }
+
   componentDidMount() {
     this._calculateData();
   }
 
+  componentWillUnmount() {
+    localStorage.setItem(this._localStorageKey, this._saveState(this.state));
```

Comment:
can we update this to use our helpers that namespace our keys?

Reply:
Ready for review on `af6c9c0`